app/src/main/java/app/revanced/integrations/shared/patches/SpoofSignaturePatch.java

```diff
@@ -0,0 +1,56 @@
+package app.revanced.integrations.shared.patches;
+
+import android.content.Context;
+import android.content.pm.PackageManager;
+
+import app.revanced.integrations.shared.settings.BaseSettings;
+import app.revanced.integrations.shared.utils.Logger;
+
+@SuppressWarnings("unused")
+public class SpoofSignaturePatch extends GmsCoreSupport {
+
+    /**
+     * Injection point.
+     */
+    public static String spoofPackageName(Context context) {
+
+        // Package name of ReVanced.
+        final String packageName = context.getPackageName();
+
+        if (!BaseSettings.SPOOF_SIGNATURE.get()) {
+            return packageName;
+        }
+
+        // Package name of YouTube or YouTube Music.
+        final String originalPackageName = getOriginalPackageName();
+
+        try {
+            PackageManager manager = context.getPackageManager();
+            manager.getPackageInfo(originalPackageName, PackageManager.GET_ACTIVITIES);
+        } catch (PackageManager.NameNotFoundException exception) {
+            Logger.printDebug(() -> "Original app was not found");
+            return packageName;
+        }
+
+        if (contentProviderClientUnAvailable(context)) {
+            Logger.printDebug(() -> "GmsCore is not running in the background");
+            return packageName;
+        }
+
+        if (batteryOptimizationsEnabled(context)) {
+            Logger.printDebug(() -> "GmsCore is not whitelisted from battery optimizations");
+            return packageName;
+        }
+
+        final String logMessage = "Package name of " + packageName + " spoofed to " + originalPackageName;
+        Logger.printDebug(() -> logMessage);
+
+        return originalPackageName;
+    }
+
+
+    // Modified by a patch. Do not touch.
+    private static String getOriginalPackageName() {
+        return "com.google.android.youtube";
+    }
+}
```

Comment:
```suggestion
package app.revanced.integrations.shared.patches;

import android.content.Context;
import android.content.pm.PackageManager;

import app.revanced.integrations.shared.settings.BaseSettings;
import app.revanced.integrations.shared.utils.Logger;

@SuppressWarnings("unused")
public class SpoofSignaturePatch extends GmsCoreSupport {

    private static final String GMS_CORE_VENDOR_GROUP_ID = getGmsCoreVendorGroupId();
    private static final String ORIGINAL_VENDOR_GROUP_ID = "com.google";

    /**
     * Injection point.
     */
    public static String spoofPackageName(Context context) {

        // Package name of ReVanced.
        final String packageName = context.getPackageName();

        if (!BaseSettings.SPOOF_SIGNATURE.get()) {
            return packageName;
        }

        // Used by the native Google app, ignored
        if (packageName.startsWith(ORIGINAL_VENDOR_GROUP_ID)) {
            Logger.printDebug(() -> "Ignored spoofing for package name: " + packageName);
            return packageName; // should never happen
        }

        if (packageName.startsWith(GMS_CORE_VENDOR_GROUP_ID) {
            return packageName.replace(GMS_CORE_VENDOR_GROUP_ID, ORIGINAL_VENDOR_GROUP_ID);
        }

        // Package name of YouTube or YouTube Music.
        final String originalPackageName = getOriginalPackageName();

        try {
            PackageManager manager = context.getPackageManager();
            manager.getPackageInfo(originalPackageName, PackageManager.GET_ACTIVITIES);
        } catch (PackageManager.NameNotFoundException exception) {
            Logger.printDebug(() -> "Original app was not found");
            return packageName;
        }
        final String logMessage = "Package name of " + packageName + " spoofed to " + originalPackageName;
        Logger.printDebug(() -> logMessage);

        return originalPackageName;
    }


    // Modified by a patch. Do not touch.
    private static String getOriginalPackageName() {
        return "com.google.android.youtube";
    }
}
```

Reply:
@inotia00 👀

Reply:
Instead of checking the RVX package name, it seems better to use [PackageSpoofUtils](https://github.com/ReVanced/GmsCore/blob/9d3ef3f11f30223a2e09e2f5800c61aad05b7222/play-services-base/core/src/main/kotlin/org/microg/gms/common/PackageSpoofUtils.kt#L100) used in GmsCore

Reply:
> There is no difference in case GmsCore conditions are not satisfied, so we can remove battery optimization check

ReVanced can run even if GmsCore battery optimizations are not disabled, so it needs to be checked